grpc-sys/grpc_wrap.cc

```diff
@@ -641,20 +641,27 @@ GPR_EXPORT grpc_call_error GPR_CALLTYPE grpcwrap_call_recv_initial_metadata(
 
 GPR_EXPORT grpc_call_error GPR_CALLTYPE grpcwrap_call_send_message(
     grpc_call* call, grpcwrap_batch_context* ctx, grpc_slice* send_buffer,
-    uint32_t write_flags, int32_t send_empty_initial_metadata, void* tag) {
+    uint32_t write_flags, grpc_metadata_array* initial_metadata, void* tag) {
   /* TODO: don't use magic number */
   grpc_op ops[2];
   memset(ops, 0, sizeof(ops));
-  size_t nops = send_empty_initial_metadata ? 2 : 1;
+  size_t nops = 1;
+
   ops[0].op = GRPC_OP_SEND_MESSAGE;
   ctx->send_message = grpc_raw_byte_buffer_create(send_buffer, 1);
   ops[0].data.send_message.send_message = ctx->send_message;
   ops[0].flags = write_flags;
   ops[0].reserved = nullptr;
-  ops[1].op = GRPC_OP_SEND_INITIAL_METADATA;
-  ops[1].flags = 0;
-  ops[1].reserved = nullptr;
 
+  if (initial_metadata) {
+    ops[nops].op = GRPC_OP_SEND_INITIAL_METADATA;
+    grpcwrap_metadata_array_move(&(ctx->send_initial_metadata), initial_metadata);
+    ops[nops].data.send_initial_metadata.count = ctx->send_initial_metadata.count;
+    ops[nops].data.send_initial_metadata.metadata = ctx->send_initial_metadata.metadata;
+    ops[nops].flags = 0;
```

Comment:
How about let caller to customize the value?

Reply:
I don't understand. Customize which value? `initial_metadata` can be full, empty, or null. If it is null, it is not sent.

Reply:
I mean `flags`. There are several flags can be set for metadata.

https://github.com/pingcap/grpc/blob/rs-release/include/grpc/impl/codegen/grpc_types.h#L493-L504

Reply:
Thanks for clarifying. I'll add `initial_metadata_flags`.

Reply:
call flags should be part of `CallOptions`, just like how header is pass.

Reply:
Okay, I did this by adding `call_flags` to `StreamingCallSink`, so that metadata on client-streaming calls can use these flags.